packages/blockchain-wallet-v4-frontend/src/modals/Btc/VerifyMessage/index.js

```diff
@@ -0,0 +1,142 @@
+// Verify the signature of a message signed with a Bitcoin address.
+
+import {
+  Banner,
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  TooltipIcon,
+  TooltipHost
+} from 'blockchain-info-components'
+
+import { FormItem, FormLabel, TextArea, TextBox } from 'components/Form'
+import * as core from './core'
+import modalEnhancer from 'providers/ModalEnhancer'
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+import { validBitcoinAddress } from 'services/FormHelper'
+
+const item = (label, input) => (
+  <FormItem style={{ marginBottom: `1rem` }}>
+    <FormLabel>
+      <div style={{ marginBottom: `0.5rem` }}>{label}</div>
+      {input}
+    </FormLabel>
+  </FormItem>
+)
+
+const ItemAddress = ({ address, network, onChange }) =>
+  item(
+    <FormattedMessage
+      id='modals.verifyMessage.address'
+      defaultMessage='Bitcoin Address:'
+    />,
+    <TextBox
+      input={{
+        onChange,
+        name: 'address'
+      }}
+      meta={{
+        error: validBitcoinAddress(address, null, { network }),
+        touched: address !== ``
+      }}
+    />
+  )
+
+const ItemMessage = ({ onChange }) =>
+  item(
+    <FormattedMessage
+      id='modals.verifyMessage.message'
+      defaultMessage='Message:'
+    />,
+    <TextArea
+      input={{
+        name: 'message',
+        onChange
+      }}
+      meta={{}}
+    />
+  )
+
+const ItemSignature = ({ onChange }) =>
+  item(
+    <FormattedMessage
+      id='modals.verifyMessage.signature'
+      defaultMessage='Signature:'
+    />,
+    <TextArea
+      input={{
+        name: 'signature',
+        onChange
+      }}
+      meta={{}}
+    />
+  )
+
+class VerifyMessage extends React.PureComponent {
+  constructor (props) {
+    super(props)
+    this.state = { address: ``, message: ``, signature: `` }
```

Comment:
This is a [React convention](https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-class) and I think important for its interaction with `this.setState`.

Reply:
the proposed variant does the same w/o a constructor, it gets compiled into the same code by babel

Reply:
Oh I see what you mean, you're referring to [class field declarations](https://github.com/tc39/proposal-class-fields).  They're only a Stage 3 proposal and may not make it into JavaScript which would mean we would have to go back and rewrite this code without them in the future.  I think it's better to stick with standard JavaScript.

Reply:
they're simplifying life a lot
also included into typescript, so doesn't seem like they're going away
we use the class fields as a part of the codebase, so it would be nice to have it uniform

Reply:
there're 0 debates around class field declarations
another example is `static` which is level 3 as well, but used by react community as a standard
also react codebase uses this approach throughout the [examples](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key)
so I'm almost completely sure this one is safe to use, and think our codebase would benefit from less constructors in a way that defining a constructor means something non-trivial going on with initial props

Reply:
OK then!